servicetalk-buffer-api/src/main/java/io/servicetalk/buffer/api/CharSequences.java

```diff
@@ -0,0 +1,346 @@
+        if (a == b) {
+            return true;
+        }
+        if (a.length() != b.length()) {
+            return false;
+        }
+        if (a.getClass() == AsciiBuffer.class) {
+            return ((AsciiBuffer) a).contentEquals(b);
+        }
+        return contentEqualsUnknownTypes(a, b);
+    }
+
+    /**
+     * Find the index of {@code c} within {@code sequence} starting at index {@code fromIndex}.
+     *
+     * @param sequence The {@link CharSequence} to search in.
+     * @param c The character to find.
+     * @param fromIndex The index to start searching (inclusive).
+     * @return The index of {@code c} or {@code -1} otherwise.
+     */
+    public static int indexOf(CharSequence sequence, char c, int fromIndex) {
+        if (sequence instanceof String) {
+            return ((String) sequence).indexOf(c, fromIndex);
+        } else if (isAsciiString(sequence)) {
+            return ((AsciiBuffer) sequence).indexOf(c, fromIndex);
+        }
+        for (int i = fromIndex; i < sequence.length(); ++i) {
+            if (sequence.charAt(i) == c) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Calculate a hash code of a byte array assuming ASCII character encoding.
+     * The resulting hash code will be case insensitive.
+     *
+     * @param seq The ascii string to produce hashcode for.
+     * @return a hashcode for the given input.
+     */
+    public static int caseInsensitiveHashCode(CharSequence seq) {
+        return isAsciiString(seq) ? seq.hashCode() : hashCodeAscii(seq);
+    }
+
+    /**
+     * Split a given {@link CharSequence} to separate ones on the given {@code delimiter}.
+     * The returned {@link CharSequence}s are created by invoking the {@link CharSequence#subSequence(int, int)} method
+     * on the main one.
+     *
+     * This method has no support for regex.
+     *
+     * @param input The initial {@link CharSequence} to split, this experiences no side effects
+     * @param delimiter The delimiter character
+     * @return a {@link List} of {@link CharSequence} subsequences of the input with the separated values
+     */
+    public static List<CharSequence> split(final CharSequence input, final char delimiter) {
+        if (input.length() == 0) {
+            return emptyList();
+        }
+
+        int startIndex = 0;
+        List<CharSequence> result = new ArrayList<>();
+        for (int i = 0; i < input.length(); i++) {
+            if (input.charAt(i) == delimiter) {
+                if ((i - startIndex) > 0) {
+                    result.add(input.subSequence(startIndex, i));
+                }
+
+                startIndex = i + 1;
+            }
+        }
+
+        if ((input.length() - startIndex) > 0) {
+            result.add(input.subSequence(startIndex, input.length()));
+        }
+        return result;
+    }
+
+    public static boolean contentEqualsIgnoreCaseUnknownTypes(final CharSequence a, final CharSequence b) {
```

Comment:
1. Does it need to be public? Can `HeaderUtils.encodingFor` use `contentEqualsIgnoreCase`? If yes, please add javadoc.
2. Can you please move it next to `contentEqualsIgnoreCase` in this class?